x-pack/plugins/elastic_assistant/server/lib/langchain/content_loaders/esql_loader.ts

```diff
@@ -37,22 +39,46 @@ export const loadESQL = async (esStore: ElasticsearchStore, logger: Logger): Pro
       true
     );
 
+    const exampleQueriesLoader = new DirectoryLoader(
+      resolve(__dirname, '../../../knowledge_base/esql/example_queries'),
+      {
+        '.asciidoc': (path) => new TextLoader(path),
+      },
+      true
+    );
+
     const docs = await docsLoader.load();
     const languageDocs = await languageLoader.load();
+    const rawExampleQueries = await exampleQueriesLoader.load();
+
+    // Add additional metadata to the example queries that indicates they are required KB documents:
+    const requiredExampleQueries = addRequiredKbResourceMetadata({
+      docs: rawExampleQueries,
+      kbResource: ESQL_RESOURCE,
+    });
```

Comment:
Side thought: perhaps another method would be to hoist these to a specific agent system message instead of retrieving each time? Not sure here, more discovery needed.... 🙂